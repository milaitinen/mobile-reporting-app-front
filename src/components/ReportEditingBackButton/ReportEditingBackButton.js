import { connect } from 'react-redux';
import React from 'react';
import { handleBack } from '../../functions/handleBack';
import { HeaderBackButton } from 'react-navigation';


/**
 *  A wrapper for the back button, that can be connected to Redux.
 *  This button is used in report editing screens so that it can show
 *  a confirmation message based on redux data (isUnsaved) and dispatch redux actions accordingly.
 */
export class ReportEditingBackButton extends React.Component {
    render() {
        return (
            <HeaderBackButton tintColor='#fff' onPress={() => handleBack(this.props.dispatch, this.props.isUnsaved)} />
        );
    }
}


// maps redux state to component props. Object that is returned can be accessed via 'this.props' e.g. this.props.email
const mapStateToProps = (state) => {
    const isUnsaved     = state.reportEditing.isUnsaved;

    return {
        isUnsaved,
    };
};

export default connect(mapStateToProps)(ReportEditingBackButton);
