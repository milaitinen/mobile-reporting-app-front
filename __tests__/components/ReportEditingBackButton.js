import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { ReportEditingBackButton } from '../../src/components/ReportEditingBackButton/ReportEditingBackButton';

configure({ adapter: new Adapter() });

test('ReportEditingBackButton', () => {

    const dispatch = jest.fn();
    const component = shallow(
        <ReportEditingBackButton dispatch={dispatch} isUnsaved={true}/>
    );

    component.find('HeaderBackButton').simulate('press');
    expect(component).toMatchSnapshot();
});