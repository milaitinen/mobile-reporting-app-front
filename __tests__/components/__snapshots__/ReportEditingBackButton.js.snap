// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportEditingBackButton 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReportEditingBackButton
    dispatch={[Function]}
    isUnsaved={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "onPress": [Function],
      "pressColorAndroid": "rgba(0, 0, 0, .32)",
      "tintColor": "#fff",
      "truncatedTitle": "Back",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "onPress": [Function],
        "pressColorAndroid": "rgba(0, 0, 0, .32)",
        "tintColor": "#fff",
        "truncatedTitle": "Back",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
